ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateArchive.java

```diff
@@ -90,20 +101,23 @@ public BonsaiWorldStateArchive(
                 blockchain, worldStateStorage, maxLayersToLoad.orElse(RETAINED_LAYERS)),
         worldStateStorage,
         blockchain,
-        useSnapshots);
+        useSnapshots,
+        metricsSystem);
   }
 
   @VisibleForTesting
   BonsaiWorldStateArchive(
       final TrieLogManager trieLogManager,
       final BonsaiWorldStateKeyValueStorage worldStateStorage,
       final Blockchain blockchain,
-      final boolean useSnapshots) {
+      final boolean useSnapshots,
+      final ObservableMetricsSystem metricsSystem) {
     this.trieLogManager = trieLogManager;
     this.blockchain = blockchain;
     this.worldStateStorage = worldStateStorage;
     this.persistedState = new BonsaiPersistedWorldState(this, worldStateStorage);
     this.useSnapshots = useSnapshots;
+    this.optimizedMerkleTrieLoader = new OptimizedMerkleTrieLoader(metricsSystem);
```

Comment:
I would suggest avoiding using new in constructor. This creates a tight, hidden coupling between classes. Inject your collaborators, throw constructor.

Reply:
done